### PR TITLE
Feature/assignment 3

### PR DESCRIPTION
**Pull request by nikhil1-successive**

### CODE REVIEW

File: src/utils/mockData.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+const nameData = ["Nikhil", "Somil", "Sanskr"]
```
Comment (shahbaz-successive) [inline src/utils/mockData.js:1]: As stated in question, create a module that means create a function that return a seeded data with logs as well such as 'Data seeding started', 'Data seeding in progress', 'Data seeding completed'

File: src/routes/userRoutes.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      secretKey,
+      { expiresIn: '10h' }
+    )
+    res.json({ token })
+  } else {
+    res.status(401).json({ message: 'Invalid username' })
+  }
+})
+
+router.get('/authorized', tokenVerificationMiddleware, (req, res) => {
```
Comment (shahbaz-successive) [inline src/routes/userRoutes.js:39]: Rename tokenVerificationMiddleware to authMiddleware

Comment (nikhil1-successive) [reply src/routes/userRoutes.js:39]: done

File: src/routes/userRoutes.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  res.json({ message: 'Welcome To Authorized Content.', user: req.user })
+})
+router.get('/console', customMiddleware, (req, res) => {
```
[the remark this answers is not in the document]
Comment (nikhil1-successive) [reply src/routes/userRoutes.js:42]: done

File: src/middleware/tokenVerificationMiddleware.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import jwt from "jsonwebtoken"
+const tokenVerificationMiddleware = (req, res, next) => {
+  const token = req.headers['authorization']
```
Comment (shahbaz-successive) [inline src/middleware/tokenVerificationMiddleware.js:3]: Define secretKey to decode token as a constant

File: src/utils/mockData.js (hunk 1 above)
Comment (nikhil1-successive) [reply src/utils/mockData.js:1]: done

File: src/routes/userRoutes.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
+import express from 'express'
+import jwt from 'jsonwebtoken'
+import customMiddleware from "../middleware/customMiddleware.js"
+import tokenVerificationMiddleware from '../middleware/tokenVerificationMiddleware.js'
```
[the remark this answers is not in the document]
Comment (nikhil1-successive) [reply src/routes/userRoutes.js]: done

File: src/middleware/tokenVerificationMiddleware.js (hunk 4 above)
Comment (nikhil1-successive) [reply src/middleware/tokenVerificationMiddleware.js]: done

**Review by shahbaz-successive — CHANGES_REQUESTED** [review]
Feedback

File: src/routes/userRoutes.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+router.post('/registerauth', authMiddleware, (req, res) => {
+  if (req.body.name) {
+    nameData.push(req.body.name)
+    res.json(nameData)
+  }
+})
+
+router.get('/login', authMiddleware, (req, res) => {
+  const { name } = req.body;
```
[the remark this answers is not in the document]
Comment (nikhil1-successive) [reply src/routes/userRoutes.js:28]: done

File: src/middleware/authMiddleware.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import jwt from "jsonwebtoken"
+const secretKey = 'Nikhil'
```
[the remark this answers is not in the document]
Comment (nikhil1-successive) [reply src/middleware/authMiddleware.js:2]: done

**Pull request merged**